README.md

```diff
@@ -2,31 +2,739 @@
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Use aggregate functions `count`, `sum`, `min`, `max`, and `average`.
+- Use the `GROUP BY` clause to group rows with the same value in one or more columns.
+- Use the `HAVING` clause to select a subset of grouped rows matching a predicate.
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+```sql
+SELECT [comma-separated list of columns and/or aggregate functions]
+FROM [table_name]
+WHERE [row selection predicate]
+GROUP BY [comma-separated list of columns]
+HAVING [aggregate function selection predicate]
+ORDER BY [comma-separated list of columns]
+LIMIT [count];
+```
 
-## Topic 1
+The basic syntax of the `SELECT` statement is enhanced to support aggregate functions and grouping:
+
+- The `SELECT` clause may include one or more aggregate function calls.
+- The `GROUP BY` clause is optional.  The clause groups sets of rows that contain the same value
+  within one or more columns.
+- The `HAVING` clause is optional. The clause selects a subset of grouped rows that match a predicate.
+
+
+## Aggregate Functions  (Code Along)
+
+We will work with the same `pet` table from the previous lesson.  
+
+SQL provides functions for aggregating values across multiple rows:
+
+- `COUNT(col)` - return a count of non-null values in column `col`.
+- `SUM(col)` - return the sum of values in column `col`.
+- `MIN(col)` - return the minimum value in column `col`.
+- `MAX(col)` - return the maximum value in column `col`.
+- `AVG(col)` - return the average value in column `col`.
+
+
+<table>
+<tr>
+<th>
+Query
+</th>
+<th>
+Result
+</th>
+</tr>
+
+<tr>
+<td>
+Count all rows:
+
+```sql
+SELECT COUNT(*)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+8
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count rows
+with a non-null value in name:
+
+```sql
+SELECT COUNT(name)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+8
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count rows
+with a non-null value in age:
+
+```sql
+SELECT COUNT(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+7
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count of distinct species:
+
+```sql
+SELECT COUNT(DISTINCT species)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+3
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count the number of dogs:
+
+```sql
+SELECT COUNT(*)
+FROM pet
+WHERE species = 'dog';
+```
+
+</td>
+<td>
+
+```text
+count
+
+4
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Sum the ages of all pets:
+
+```sql
+SELECT SUM(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+sum
+
+35
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Sum the ages of Tabby cats:
+
+```sql
+SELECT SUM(age)
+FROM pet
+WHERE breed = 'Tabby' and species = 'cat' ;
+```
+
+</td>
+<td>
+
+```text
+sum
+
+11
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Minimum, maximum and average age
+among all pets:
+
+```sql
+SELECT MIN(age), MAX(age), AVG(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   12  5    
```

Comment:
Shouldn't the average be a decimal? Like 5.428571428?

Reply:
you are correct, not sure why it truncated.